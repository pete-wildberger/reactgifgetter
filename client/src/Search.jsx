import React, { Component } from 'react';
import axios from 'axios';
// import GifCard from './GifCard';
import Header from './Header';

// import preload from '../data.json';

class Search extends Component {
  constructor(props) {
    super(props);
    this.state = {
      searchTerm: '',
      apiData: [],
      apiObj: {}
    };
    this.handleSearchTermChange = this.handleSearchTermChange.bind(this);
    this.findGifs = this.findGifs.bind(this);
  }

  componentDidMount() {
    axios.get('http://api.giphy.com/v1/gifs/random?api_key=dc6zaTOxFJmzC&tag=').then(response => {
      console.log(response);
      console.log(this.state);
      const posts = response.data.data;
      this.setState({ apiObj: posts });
    });
  }
  handleSearchTermChange = event => {
    this.setState({ searchTerm: event.target.value });
  };
  findGifs(term) {
    let searchURL = `http://api.giphy.com/v1/gifs/search?q=${term}&api_key=dc6zaTOxFJmzC`;
    axios.get(searchURL).then(response => {
      console.log('response ', response);
      this.setState({ apiData: response.data.data, apiObj: {} });
      console.log('Search.state ', this.state);
    });
  }
  displayGif() {
    let gif;
    let gifs;
    if (this.state.apiData.length > 0) {
      gifs = this.state.apiData;
      return gifs.map(gif => {
        return (
          <div key={gif.id}>
            <img src={gif.images.downsized.url} alt={gif.images.downsized.url} />
          </div>
        );
      });
    }
    if (this.state.apiObj != {}) {
      gif = this.state.apiObj;
      return (
        <div key={gif.image_url}>
          <img src={gif.image_url} alt={gif.image_url} />
        </div>
      );
    }
  }
  render() {
    return (
      <div className="search">
        <Header searchTerm={this.state.searchTerm} showSearch handleSearchTermChange={this.handleSearchTermChange} />
        <button onClick={() => this.findGifs(this.state.searchTerm)}>search</button>
        <div>{this.displayGif()}</div>
      </div>
    );
  }
}

export default Search;
