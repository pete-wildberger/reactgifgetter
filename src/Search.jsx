// @flow
import React, { Component } from 'react';
import axios from 'axios';
import GifCard from './GifCard';
import Header from './Header';

// import preload from '../data.json';

class Search extends Component {
  // constructor(props) {
  //   super(props);
  //   this.state = {
  //     searchTerm: 'this is a debug statement'
  //   };
  //   this.handleSearchTermChange = this.handleSearchTermChange.bind(this);
  // }
  state = {
    searchTerm: '',
    apiData: [],
    apiObj: {}
  };

  componentDidMount() {
    axios.get('http://api.giphy.com/v1/gifs/random?api_key=dc6zaTOxFJmzC&tag=').then(response => {
      console.log(response);
      console.log(this.state);
      const posts = response.data.data;
      this.setState({ apiObj: posts });
    });
  }
  handleSearchTermChange = event => {
    this.setState({ searchTerm: event.target.value });
  };
  findGifs(term) {
    let searchURL = 'http://api.giphy.com/v1/gifs/search?q=';
    searchURL += term;
    searchURL += '&api_key=dc6zaTOxFJmzC';
    axios.get(searchURL).then(response => {
      console.log(response);
      console.log(this.state);
      const posts = response.data.data;
      this.setState({ apiData: posts, apiObj: {} });
    });
  }
  render() {
    let gif;
    if (this.state.apiObj != {}) {
      gif = this.state.apiObj;
    } else {
      gif = this.state.apiData;
    }
    // const items = this.props.items.map(item => (
    //   <ul key={item.id}>
    //     <li>
    //       <button onClick={() => this.displayAlert(item.email)}>{item.lastName + ', ' + item.firstName}</button>
    //     </li>
    //   </ul>
    // ));
    return (
      <div className="search">
        <Header searchTerm={this.state.searchTerm} showSearch handleSearchTermChange={this.handleSearchTermChange} />
        <button onClick={() => this.findGifs(this.state.searchTerm)}>search</button>
        <div>
          <GifCard key={gif.img_url} {...gif} />
        </div>
      </div>
    );
  }
}
export default Search;
