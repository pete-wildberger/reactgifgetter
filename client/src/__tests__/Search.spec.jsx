import React from 'react';
import ReactTestUtils from 'react-dom';
import { shallow } from 'enzyme';
import Search from '../components/Search.jsx';

test('Search renders correctly', () => {
  const component = shallow(<Search />);
  expect(component).toMatchSnapshot();
});

// test('Search renders correctly', () => {
//   const component = shallow(<Search />);
//   expect(component).toMatchSnapshot();
// });
