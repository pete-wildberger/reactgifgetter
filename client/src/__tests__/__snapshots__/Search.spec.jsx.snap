// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`Search renders correctly 1`] = `
ShallowWrapper {
  "length": 1,
  Symbol(enzyme.__root__): [Circular],
  Symbol(enzyme.__unrendered__): <Search />,
  Symbol(enzyme.__renderer__): Object {
    "batchedUpdates": [Function],
    "getNode": [Function],
    "render": [Function],
    "simulateEvent": [Function],
    "unmount": [Function],
  },
  Symbol(enzyme.__node__): Object {
    "instance": null,
    "key": undefined,
    "nodeType": "host",
    "props": Object {
      "children": Array [
        <Header
          handleSearchTermChange={[Function]}
          searchTerm=""
          showSearch={true}
        />,
        <button
          onClick={[Function]}
        >
          Search Gifs
        </button>,
        <button
          onClick={[Function]}
        >
          Random Gif
        </button>,
        <Link
          replace={false}
          to="/faves"
        >
          <button>
            Faves
          </button>
        </Link>,
        <div
          className="row"
        >
          <div
            className="col-3"
          >
            <img
              alt={undefined}
              src={undefined}
            />
          </div>
        </div>,
        <Footer />,
      ],
      "className": "app",
    },
    "ref": null,
    "rendered": Array [
      Object {
        "instance": null,
        "key": undefined,
        "nodeType": "function",
        "props": Object {
          "handleSearchTermChange": [Function],
          "searchTerm": "",
          "showSearch": true,
        },
        "ref": null,
        "rendered": null,
        "type": [Function],
      },
      Object {
        "instance": null,
        "key": undefined,
        "nodeType": "host",
        "props": Object {
          "children": "Search Gifs",
          "onClick": [Function],
        },
        "ref": null,
        "rendered": "Search Gifs",
        "type": "button",
      },
      Object {
        "instance": null,
        "key": undefined,
        "nodeType": "host",
        "props": Object {
          "children": "Random Gif",
          "onClick": [Function],
        },
        "ref": null,
        "rendered": "Random Gif",
        "type": "button",
      },
      Object {
        "instance": null,
        "key": undefined,
        "nodeType": "class",
        "props": Object {
          "children": <button>
            Faves
          </button>,
          "replace": false,
          "to": "/faves",
        },
        "ref": null,
        "rendered": Object {
          "instance": null,
          "key": undefined,
          "nodeType": "host",
          "props": Object {
            "children": "Faves",
          },
          "ref": null,
          "rendered": "Faves",
          "type": "button",
        },
        "type": [Function],
      },
      Object {
        "instance": null,
        "key": undefined,
        "nodeType": "host",
        "props": Object {
          "children": <div
            className="col-3"
          >
            <img
              alt={undefined}
              src={undefined}
            />
          </div>,
          "className": "row",
        },
        "ref": null,
        "rendered": Object {
          "instance": null,
          "key": undefined,
          "nodeType": "host",
          "props": Object {
            "children": <img
              alt={undefined}
              src={undefined}
            />,
            "className": "col-3",
          },
          "ref": null,
          "rendered": Object {
            "instance": null,
            "key": undefined,
            "nodeType": "host",
            "props": Object {
              "alt": undefined,
              "src": undefined,
            },
            "ref": null,
            "rendered": null,
            "type": "img",
          },
          "type": "div",
        },
        "type": "div",
      },
      Object {
        "instance": null,
        "key": undefined,
        "nodeType": "function",
        "props": Object {},
        "ref": null,
        "rendered": null,
        "type": [Function],
      },
    ],
    "type": "div",
  },
  Symbol(enzyme.__nodes__): Array [
    Object {
      "instance": null,
      "key": undefined,
      "nodeType": "host",
      "props": Object {
        "children": Array [
          <Header
            handleSearchTermChange={[Function]}
            searchTerm=""
            showSearch={true}
          />,
          <button
            onClick={[Function]}
          >
            Search Gifs
          </button>,
          <button
            onClick={[Function]}
          >
            Random Gif
          </button>,
          <Link
            replace={false}
            to="/faves"
          >
            <button>
              Faves
            </button>
          </Link>,
          <div
            className="row"
          >
            <div
              className="col-3"
            >
              <img
                alt={undefined}
                src={undefined}
              />
            </div>
          </div>,
          <Footer />,
        ],
        "className": "app",
      },
      "ref": null,
      "rendered": Array [
        Object {
          "instance": null,
          "key": undefined,
          "nodeType": "function",
          "props": Object {
            "handleSearchTermChange": [Function],
            "searchTerm": "",
            "showSearch": true,
          },
          "ref": null,
          "rendered": null,
          "type": [Function],
        },
        Object {
          "instance": null,
          "key": undefined,
          "nodeType": "host",
          "props": Object {
            "children": "Search Gifs",
            "onClick": [Function],
          },
          "ref": null,
          "rendered": "Search Gifs",
          "type": "button",
        },
        Object {
          "instance": null,
          "key": undefined,
          "nodeType": "host",
          "props": Object {
            "children": "Random Gif",
            "onClick": [Function],
          },
          "ref": null,
          "rendered": "Random Gif",
          "type": "button",
        },
        Object {
          "instance": null,
          "key": undefined,
          "nodeType": "class",
          "props": Object {
            "children": <button>
              Faves
            </button>,
            "replace": false,
            "to": "/faves",
          },
          "ref": null,
          "rendered": Object {
            "instance": null,
            "key": undefined,
            "nodeType": "host",
            "props": Object {
              "children": "Faves",
            },
            "ref": null,
            "rendered": "Faves",
            "type": "button",
          },
          "type": [Function],
        },
        Object {
          "instance": null,
          "key": undefined,
          "nodeType": "host",
          "props": Object {
            "children": <div
              className="col-3"
            >
              <img
                alt={undefined}
                src={undefined}
              />
            </div>,
            "className": "row",
          },
          "ref": null,
          "rendered": Object {
            "instance": null,
            "key": undefined,
            "nodeType": "host",
            "props": Object {
              "children": <img
                alt={undefined}
                src={undefined}
              />,
              "className": "col-3",
            },
            "ref": null,
            "rendered": Object {
              "instance": null,
              "key": undefined,
              "nodeType": "host",
              "props": Object {
                "alt": undefined,
                "src": undefined,
              },
              "ref": null,
              "rendered": null,
              "type": "img",
            },
            "type": "div",
          },
          "type": "div",
        },
        Object {
          "instance": null,
          "key": undefined,
          "nodeType": "function",
          "props": Object {},
          "ref": null,
          "rendered": null,
          "type": [Function],
        },
      ],
      "type": "div",
    },
  ],
  Symbol(enzyme.__options__): Object {
    "adapter": ReactSixteenAdapter {
      "options": Object {
        "enableComponentDidUpdateOnSetState": true,
      },
    },
  },
}
`;
